File: frontend/apps/ios/Pantopus/Features/BusinessProfile/Components/ServicesList.swift
//
//  ServicesList.swift
//  Pantopus
//
//  A10.6 — the Services card: priced rows with a business-tinted glyph
//  tile, name + meta, and a right-aligned price + unit.
//
//  Design reference: `docs/designs/A10/business-frames.jsx` (Services).
//

import SwiftUI

@MainActor
struct ServicesList: View {
    let services: [BusinessServiceRow]

    var body: some View {
        VStack(spacing: Spacing.s0) {
            ForEach(Array(services.enumerated()), id: \.element.id) { index, service in
                row(service)
                if index != services.count - 1 {
                    Rectangle()
                        .fill(Theme.Color.appBorderSubtle)
                        .frame(height: 1)
                        .padding(.horizontal, 14)
                }
            }
        }
        .background(Theme.Color.appSurface)
        .clipShape(RoundedRectangle(cornerRadius: Radii.lg, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: Radii.lg, style: .continuous)
                .stroke(Theme.Color.appBorder, lineWidth: 1)
        )
        .accessibilityIdentifier("businessProfile.services")
    }

    private func row(_ service: BusinessServiceRow) -> some View {
        HStack(spacing: Spacing.s3) {
            ZStack {
                RoundedRectangle(cornerRadius: Radii.md, style: .continuous)
                    .fill(Theme.Color.businessBg)
                    .frame(width: 34, height: 34)
                Icon(service.icon, size: 16, strokeWidth: 2, color: Theme.Color.business)
            }
            VStack(alignment: .leading, spacing: 1) {
                Text(service.name)
                    .font(.system(size: 13, weight: .semibold))
                    .tracking(-0.1)
                    .foregroundStyle(Theme.Color.appText)
                if let detail = service.detail, !detail.isEmpty {
                    Text(detail)
                        .font(.system(size: 11))
                        .foregroundStyle(Theme.Color.appTextSecondary)
                }
            }
            Spacer(minLength: Spacing.s2)
            VStack(alignment: .trailing, spacing: 0) {
                Text(service.priceLabel)
                    .font(.system(size: 13, weight: .bold))
                    .tracking(-0.2)
                    .foregroundStyle(Theme.Color.appText)
                if let unit = service.unit, !unit.isEmpty {
                    Text(unit)
                        .font(.system(size: 10))
                        .foregroundStyle(Theme.Color.appTextMuted)
                }
            }
        }
        .padding(.horizontal, 14)
        .padding(.vertical, 12)
        .accessibilityElement(children: .combine)
        .accessibilityLabel("\(service.name), \(service.priceLabel)")
    }
}

#Preview("ServicesList") {
    ServicesList(services: [
        BusinessServiceRow(
            id: "std",
            name: "Standard clean",
            detail: "2 hr · 2-person team",
            priceLabel: "from $90",
            unit: "per visit",
            icon: .droplets
        ),
        BusinessServiceRow(
            id: "deep",
            name: "Deep clean",
            detail: "4 hr · baseboards, inside oven",
            priceLabel: "from $180",
            unit: "per visit",
            icon: .sparkles
        )
    ])
    .padding(Spacing.s4)
    .background(Theme.Color.appBg)
}

File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/business_profile/components/ServicesList.kt
@file:Suppress("MagicNumber", "PackageNaming")

package app.pantopus.android.ui.screens.business_profile.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.pantopus.android.ui.screens.business_profile.BusinessServiceRow
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusIconImage
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing

/**
 * A10.6 — the Services card: priced rows with a business-tinted glyph
 * tile, name + meta, and a right-aligned price + unit.
 *
 * Mirror of iOS `Features/BusinessProfile/Components/ServicesList.swift`.
 */
@Composable
fun ServicesList(
    services: List<BusinessServiceRow>,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.lg))
                .border(1.dp, PantopusColors.appBorder, RoundedCornerShape(Radii.lg))
                .background(PantopusColors.appSurface)
                .testTag("businessProfile.services"),
    ) {
        services.forEachIndexed { index, service ->
            ServiceRow(service)
            if (index != services.lastIndex) {
                HorizontalDivider(
                    color = PantopusColors.appBorderSubtle,
                    modifier = Modifier.padding(horizontal = 14.dp),
                )
            }
        }
    }
}

@Composable
private fun ServiceRow(service: BusinessServiceRow) {
    Row(
        modifier =
            Modifier
                .fillMaxWidth()
                .padding(horizontal = 14.dp, vertical = 12.dp)
                .semantics { contentDescription = "${service.name}, ${service.priceLabel}" },
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(Spacing.s3),
    ) {
        Box(
            modifier =
                Modifier
                    .size(34.dp)
                    .clip(RoundedCornerShape(Radii.md))
                    .background(PantopusColors.businessBg),
            contentAlignment = Alignment.Center,
        ) {
            PantopusIconImage(
                icon = service.icon,
                contentDescription = null,
                size = 16.dp,
                strokeWidth = 2f,
                tint = PantopusColors.business,
            )
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = service.name,
                color = PantopusColors.appText,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                letterSpacing = (-0.1).sp,
            )
            if (!service.detail.isNullOrEmpty()) {
                Text(
                    text = service.detail,
                    color = PantopusColors.appTextSecondary,
                    fontSize = 11.sp,
                )
            }
        }
        Column(horizontalAlignment = Alignment.End) {
            Text(
                text = service.priceLabel,
                color = PantopusColors.appText,
                fontSize = 13.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = (-0.2).sp,
            )
            if (!service.unit.isNullOrEmpty()) {
                Text(
                    text = service.unit,
                    color = PantopusColors.appTextMuted,
                    fontSize = 10.sp,
                )
            }
        }
    }
}
